widgetssdk/src/main/java/com/glia/widgets/core/survey/GliaSurveyRepository.java

```diff
@@ -0,0 +1,54 @@
+package com.glia.widgets.core.survey;
+
+import androidx.annotation.NonNull;
+
+import com.glia.androidsdk.Engagement;
+import com.glia.androidsdk.GliaException;
+import com.glia.androidsdk.RequestCallback;
+import com.glia.androidsdk.engagement.Survey;
+import com.glia.widgets.di.Dependencies;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Consumer;
+
+public class GliaSurveyRepository implements RequestCallback<Survey> {
+
+    private Engagement engagement;
+    private final List<OnSurveyListener> listeners = new ArrayList<>();
+
+    public void onEngagementEnded(Engagement engagement) {
+        if (listeners.isEmpty()) { // No action needs if the listeners are empty.
+            return;
+        }
+        if (this.engagement != null) { // Prevent multiple surveys fetching.
```

Comment:
subscribe on resume and unsubscribe on pause should solve this issue

Reply:
Just a thought: Is it necessary to fetch the survey in the repository? Fetching could be done in the usecase itself and then only usecase instance creator get's the result.